jina/executors/indexers/keyvalue.py

```diff
@@ -75,6 +75,25 @@ def query(self, key: int) -> Optional[bytes]:
             with mmap.mmap(self.query_handler.body, offset=p, length=l) as m:
                 return m[r:]
 
+    def update(self, keys: Iterator[int], values: Iterator[bytes], *args, **kwargs):
+        for key, new_value in zip(keys, values):
+            pos_info = self.query_handler.header.get(key, None)
+            p, r, l = pos_info
```

Comment:
Maybe just something more explicit on what `p, r, l `are?

Reply:
fun fact: when I wrote something like this (often intentionally), it's usually a sign this code is hardcore and at low-level, to remind you Data Structure & Algorithms 101